src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -338,33 +298,125 @@ private function getRelationDetailsFromInput($input)
 
         $relationDetails = [];
         foreach ($relationFields as $field) {
-            // we split the entity into relations, eg: user.accountDetails.address
-            // (user -> HasOne accountDetails -> BelongsTo address)
-            // we specifically use only the relation entity because relations like
-            // HasOne and MorphOne use the attribute in the relation string
-            $key = implode('.relations.', explode('.', $this->getOnlyRelationEntity($field)));
-            $attributeName = (string) Str::of($field['name'])->afterLast('.');
+            $relationDetails = $this->geFieldDetailsForRelationSaving($field, $input, $relationDetails);
+
+            if (isset($field['subfields'])) {
+                foreach ($field['subfields'] as $subfield) {
+                    $subfield['baseModel'] = $field['model'];
+                    $subfield['baseEntity'] = $field['entity'];
+                    $subfield = $this->makeSureFieldHasNecessaryAttributes($subfield);
+                    if (isset($subfield['relation_type'])) {
+                        $relationDetails = $this->geFieldDetailsForRelationSaving($subfield, $input, $relationDetails, $field);
+                    }
+                }
+            }
+        }
+
+        return $relationDetails;
+    }
+
+    private function geFieldDetailsForRelationSaving($field, $input, $relationDetails, $parent = false)
```

Comment:
This is the HEAVY change in this PR. 

We will discuss this better tomorrow, if you have other stuff to look at, please ignore code review in this function as I am going to refactor it in the morning.

Reply:
Ok I'll wait. Just from seeing the HUGE id/else statement I can tell that the interior of this should probably become 2 separate functions or something.